routing/endpointregistry.go

```diff
@@ -103,12 +148,31 @@ func (r *EndpointRegistry) Do(routes []*Route) []*Route {
 	}
 
 	removeOlder := now.Add(-r.lastSeenTimeout)
+	resetOlder := now.Add(-r.statsResetPeriod)
 	r.data.Range(func(key, value any) bool {
 		e := value.(*entry)
 		if e.LastSeen().Before(removeOlder) {
 			r.data.Delete(key)
 		}
 
+		if r.passiveHealthCheckEnabled && e.lastStatsResetTime().Before(resetOlder) {
+			e.setLastStatsReset(now)
+			curSlot := e.curSlot.Load()
+			nextSlot := (curSlot + 1) % 2
+			e.totalFailedRoundTrips[nextSlot].Store(0)
+			e.totalRequests[nextSlot].Store(0)
```

Comment:
@AlexanderYastrebov the basic idea is that one slot is to write current observations and the other is to use for reading for the hot traffic algorithm. If we swap the slot every time.Duration (think "1s"), then our PHC will have reaction time of 1s.

Reply:
I would kindly ask to describe this race in more detail, because I was taking quite a lot of looks, and I don't see the fatal scenarios at the moment.

Reply:
Discussed in person, we think it's a good approach, maybe premature optimization, but it's the fast data-path solution.